drasyl-extras/src/main/java/org/drasyl/handler/connection/ConnectionHandler.java

```diff
@@ -0,0 +1,2769 @@
+    static final ConnectionAbortedDueToUserTimeoutException CONNECTION_USER_TIMEOUT_EXCEPTION = new ConnectionAbortedDueToUserTimeoutException();
+    private final ConnectionConfig config;
+    State state;
+    TransmissionControlBlock tcb;
+    ScheduledFuture<?> userTimer;
+    ScheduledFuture<?> retransmissionTimer;
+    ScheduledFuture<?> timeWaitTimer;
+    ScheduledFuture<?> zeroWindowProber;
+    private ChannelPromise establishedPromise;
+    private ChannelPromise closedPromise;
+    private boolean readPending;
+    private ChannelHandlerContext ctx;
+    private ChannelPromise segmentizedFuture;
+    private long segmentizedRemainingBytes;
+
+    @SuppressWarnings("java:S107")
+    ConnectionHandler(final ConnectionConfig config,
+                      final State state,
+                      final TransmissionControlBlock tcb,
+                      final ScheduledFuture<?> userTimer,
+                      final ScheduledFuture<?> retransmissionTimer,
+                      final ScheduledFuture<?> timeWaitTimer,
+                      final ChannelPromise establishedPromise,
+                      final ChannelPromise closedPromise,
+                      final ChannelHandlerContext ctx) {
+        this.config = requireNonNull(config);
+        this.state = state;
+        this.tcb = tcb;
+        this.userTimer = userTimer;
+        this.retransmissionTimer = retransmissionTimer;
+        this.timeWaitTimer = timeWaitTimer;
+        this.establishedPromise = establishedPromise;
+        this.closedPromise = closedPromise;
+        this.ctx = ctx;
+    }
+
+    public ConnectionHandler(final ConnectionConfig config) {
+        this(config, null, null, null, null, null, null, null, null);
+    }
+
+    public ConnectionHandler() {
+        this(ConnectionConfig.DEFAULT);
+    }
+
+    @Override
+    public String toString() {
+        return "ConnectionHandler{" +
+                "state=" + state +
+                ", tcb=" + tcb +
+                '}';
+    }
+
+    /*
+     * Handler Events
+     */
+
+    @Override
+    public void handlerAdded(final ChannelHandlerContext ctx) {
+        this.ctx = ctx;
+        if (ctx.channel().isActive()) {
+            initHandler(ctx);
+        }
+    }
+
+    @Override
+    public void handlerRemoved(final ChannelHandlerContext ctx) {
+        // release all resources/timers
+        deleteTcb();
+        cancelUserTimer(ctx);
+        cancelRetransmissionTimer(ctx);
+        cancelTimeWaitTimer(ctx);
+        establishedPromise.tryFailure(CONNECTION_CLOSING_ERROR);
+        ctx.channel().closeFuture().addListener(new PromiseNotifier<>(false, closedPromise));
+    }
+
+    /*
+     * Channel Signals
+     */
+
+    @Override
+    public void close(final ChannelHandlerContext ctx,
+                      final ChannelPromise promise) {
+        userCallClose(ctx, promise);
```

Comment:
Warum wird das close nicht in der Pipeline weitergegeben?

Reply:
Weil erst der teardown handshake abgeschlossen werden soll